include/cppkafka/kafka_handle_base.h

```diff
@@ -357,7 +381,20 @@ class CPPKAFKA_API KafkaHandleBase {
 private:
     static const std::chrono::milliseconds DEFAULT_TIMEOUT;
 
-    using HandlePtr = std::unique_ptr<rd_kafka_t, decltype(&rd_kafka_destroy)>;
+    // It seems that destroy_flags maybe RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE by default.
+    //
+    // All the consumer objects created by cppkafka call rd_kafka_consumer_close during destruction.
+    // So we don't want it to be called once again during handle destruction.
+    int destroy_flags_ = 0;
+
+    struct handle_deleter {
```

Comment:
Use camel case for class names i.e. `HandleDeleter`. 
Make constructor `explicit`. 
I would change to `const KafkaHandleBase& handle_base_` because having this as nullptr is not valid.